test/junit/scala/collection/mutable/ArrayBufferTest.scala

```diff
@@ -455,4 +455,21 @@ class ArrayBufferTest {
     buf.insertAll(1, buf)
     assertSameElements(List(1, 1, 2, 3, 2, 3), buf)
   }
+
+  // scala/bug#12284
+  @Test
+  def viewConsistency(): Unit = {
+    def check[U](op: ArrayBuffer[Int] => U): Unit = {
+      val buf = ArrayBuffer(1, 2, 3)
```

Comment:
could we instead use scalacheck here? It's not clear there aren't edge cases at empty, single element, etc...
```scala
forAll(genArrayBuffer) { buf =>
 ...
}
```

Reply:
what would the code after that be? sorry, I'm not super familiar with scalacheck, and I'm not following your suggestion

Reply:
```scala
forAll { (ints: List[Int]) =>
  val buf = ArrayBuffer(ints: _*)
  
  def check(op: ArrayBuffer[Int] => Any): Unit = {
    val view = buf.view
    op(buf)
    assertSameElements(buf, view)
  }

  check(_.clear())
  check(_dropRightInPlace(1))
  check(_.dropInPlace(1))
  check(_ ++= (1 to 100))
  check (_.insertAll(1, 1 to 100)) 
}
```

IME scalacheck is a massive win on bug finding. I virtually always find bugs when I add scalacheck tests.

Reply:
ftr, this is missing a bunch of stuff, including:

- `assertSameElements` should just be a call to `sameElements` (or `==`), so that it's a `Boolean` rather than something that throws
- the calls to `check` need to be combined (using `&&`? `++`?) so that all but the last aren't ignored
- `insertAll(1, ???)` will throw an exception if the `ArrayBuffer` is empty
- `++=` and `insertAll` will throw an exception if the `ArrayBuffer` is larger than `Int.MaxValue / 2 - 100` (that value is based on the implementation, and theoretically should allow 

relating to that second point, I was looking at some other classes to try to understand how scalacheck works, and it appears that there are other places that drop checks on the floor, including https://github.com/scala/scala/blob/66e8d34e646a616f39615cbf7e8fc6d56103b2ec/test/scalacheck/scala/collection/immutable/VectorMapProperties.scala#L38-L42

Reply:
I've resolved this (added scalacheck property tests), but I'm gonna leave it unresolved for now as a reminder that some other property tests are broken

Reply:
regarding `VectorMapProperties`: see #9768